src/MSRIOGroup.cpp

```diff
@@ -285,6 +288,82 @@ namespace geopm
         }
     }
 
+    void MSRIOGroup::register_rdt_signals(void)
+    {
+        // It may be necessary to wrap the initial rdt() call in this check
+        int domain = signal_domain_type("MSR::MISC_ENABLE:LIMIT_CPUID_MAXVAL");
+        int num_domain = m_platform_topo.num_domain(domain);
+        double disable = 0.0;
+        for (int dom_idx = 0; dom_idx < num_domain; ++dom_idx) {
+            disable += read_signal("MSR::MISC_ENABLE:LIMIT_CPUID_MAXVAL", domain, dom_idx);
```

Comment:
If we use agg function of and logical_and this can be done with a single read_signal call.